superset-frontend/src/SqlLab/components/AceEditorWrapper/useAnnotations.test.ts

```diff
@@ -0,0 +1,237 @@
+import { initialState } from 'src/SqlLab/fixtures';
+import COMMON_ERR_MESSAGES from 'src/utils/errorMessages';
+import { useAnnotations } from './useAnnotations';
+
+const fakeApiResult = {
+  result: [
+    {
+      end_column: null,
+      line_number: 3,
+      message: 'ERROR: syntax error at or near ";"',
+      start_column: null,
+    },
+  ],
+};
+const expectDbId = 'db1';
+const expectSchema = 'my_schema';
+const expectSql = 'SELECT * from example_table';
+const expectTemplateParams = '{"a": 1, "v": "str"}';
+const expectValidatorEngine = 'defined_validator';
+const queryValidationApiRoute = `glob:*/api/v1/database/${expectDbId}/validate_sql/`;
+
+jest.mock('@superset-ui/core', () => ({
+  ...jest.requireActual('@superset-ui/core'),
+  t: (str: string) => str,
+}));
+
+afterEach(() => {
+  fetchMock.reset();
+  act(() => {
+    store.dispatch(api.util.resetApiState());
+  });
+});
+
+beforeEach(() => {
+  fetchMock.post(queryValidationApiRoute, fakeApiResult);
+});
+
+test('skips fetching validation if validator is undefined', () => {
+  const { result } = renderHook(
+    () =>
+      useAnnotations({
+        sql: expectSql,
+        dbId: expectDbId,
+        schema: expectSchema,
+        templateParams: expectTemplateParams,
+      }),
+    {
+      wrapper: createWrapper({
+        useRedux: true,
+        store,
+      }),
+    },
+  );
+  expect(result.current.data).toEqual([]);
+  expect(fetchMock.calls(queryValidationApiRoute)).toHaveLength(0);
+});
+
+test('returns validation if validator is configured', async () => {
+  const { result, waitFor } = renderHook(
+    () =>
+      useAnnotations({
+        sql: expectSql,
+        dbId: expectDbId,
+        schema: expectSchema,
+        templateParams: expectTemplateParams,
+      }),
+    {
+      wrapper: createWrapper({
+        useRedux: true,
+        initialState: {
+          ...initialState,
+          sqlLab: {
+            ...initialState.sqlLab,
+            databases: {
+              [expectDbId]: {
+                backend: expectValidatorEngine,
+              },
+            },
+          },
+          common: {
+            conf: {
+              SQL_VALIDATORS_BY_ENGINE: {
+                [expectValidatorEngine]: true,
+              },
+            },
+          },
+        },
+      }),
+    },
+  );
+  await waitFor(() =>
+    expect(fetchMock.calls(queryValidationApiRoute)).toHaveLength(1),
+  );
+  expect(result.current.data).toEqual(
+    fakeApiResult.result.map(err => ({
+      type: 'error',
+      row: (err.line_number || 0) - 1,
+      column: (err.start_column || 0) - 1,
+      text: err.message,
+    })),
+  );
+});
+
+test('returns server error description', async () => {
+  const errorMessage = 'Unexpected validation api error';
+  fetchMock.post(
+    queryValidationApiRoute,
+    {
+      throws: new Error(errorMessage),
+    },
+    { overwriteRoutes: true },
+  );
+  const { result, waitFor } = renderHook(
```

Comment:
This piece of code is identical to previous tests. Can you extract a function called `initialize` that accepts an optional `withValidator`? Something like:

```
const initialize = (withValidator?: boolean) => {...};

// First test
const { result } = initialize() 

// Second and third tests
const { result } = initialize(true) 
```

Reply:
Done!